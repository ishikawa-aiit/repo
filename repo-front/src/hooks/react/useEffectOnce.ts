import { useEffect, useRef, useState } from 'react'

/**
 * 依存関係がゼロの時、useEffect（）の代わりに使用するカスタムフック
 *
 * @remarks
 * このカスタムフックはReact 18の仕様変更のため、作成しています（React 18では依存関係がゼロの時、2回レンダリングされる）
 * https://github.com/vercel/next.js/issues/35822
 */
export const useEffectOnce = (effect: () => void | (() => void)) => {
  const destroyFunc = useRef<void | (() => void)>()
  const effectCalled = useRef(false)
  const renderAfterCalled = useRef(false)
  const [_val, setVal] = useState<number>(0)

  if (effectCalled.current) {
    renderAfterCalled.current = true
  }

  useEffect(() => {
    if (!effectCalled.current) {
      destroyFunc.current = effect()
      effectCalled.current = true
    }

    setVal((val) => {
      return val + 1
    })

    return () => {
      if (!renderAfterCalled.current) {
        return
      }
      if (destroyFunc.current) {
        destroyFunc.current()
      }
    }
    // マウント時に1回だけ実行するため
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])
}
