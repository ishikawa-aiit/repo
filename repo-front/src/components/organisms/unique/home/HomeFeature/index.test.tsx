import { render } from '@testing-library/react'

import { HomeFeature } from '.'

describe('snapshots', () => {
  it('最新のスナップショットに更新されている', () => {
    const { container } = render(<HomeFeature />)

    expect(container).toMatchSnapshot()
  })
})
