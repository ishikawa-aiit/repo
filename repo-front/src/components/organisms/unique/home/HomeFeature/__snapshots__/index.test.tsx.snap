// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 最新のスナップショットに更新されている 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-0::before {
  display: block;
  content: "";
  width: 688px;
  height: 477px;
  position: absolute;
  top: -52px;
  right: 0px;
  background: url("/images/pages/home/bg_right-obj_01.svg") no-repeat 0 0;
  z-index: -1;
}

.emotion-1 {
  width: 1264px;
  margin: 0 auto;
}

.emotion-2 {
  font-size: 30px;
  margin-top: 120px;
  margin-bottom: 50px;
  padding-bottom: 6px;
  text-align: center;
  position: relative;
}

.emotion-2::after {
  display: block;
  content: "";
  width: 4rem;
  height: 4px;
  background-color: primary.500;
  border-radius: 9999px;
  position: absolute;
  left: 50%;
  bottom: 0px;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-3 {
  display: grid;
  grid-template-columns: repeat(3, minmax(0, 1fr));
  margin-bottom: 40px;
}

.emotion-4 {
  padding: 8px;
}

.emotion-5 {
  margin-bottom: 6px;
}

.emotion-7 {
  padding: 8px;
  border-left: 1px;
  border-color: neutral.200;
}

<div>
  <div
    class="emotion-0"
    data-cy="HomeFeature"
    data-testid="HomeFeature"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="chakra-heading emotion-2"
      >
        Repoの特徴
      </h2>
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          <h3
            class="chakra-heading emotion-5"
          >
            簡単に過去の週報を見返せる！
          </h3>
          <p
            class="chakra-text emotion-6"
          >
             
            週報一覧から参照したい週報を容易にストレスなく閲覧することができます。
          </p>
        </div>
        <div
          class="emotion-7"
        >
          <h3
            class="chakra-heading emotion-5"
          >
            週報を素早く記入できる！
          </h3>
          <p
            class="chakra-text emotion-6"
          >
             
            週報作成時に登録したフォーマットを反映させることで、入力時間を大幅に短縮できます。
          </p>
        </div>
        <div
          class="emotion-7"
        >
          <h3
            class="chakra-heading emotion-5"
          >
            週報の記入を忘れない！
          </h3>
          <p
            class="chakra-text emotion-6"
          >
             
            週報提出期限が近づくと、リマインダーを受け取ることができます。
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
